src/ex_color/model.py

```diff
@@ -1,16 +1,37 @@
 import logging
 
+import lightning as L
 import torch
 import torch.nn as nn
+import torch.optim as optim
+from torch import Tensor
+
+from ex_color.criteria.criteria import LossCriterion, RegularizerConfig
+from ex_color.result import InferenceResult
+from mini.temporal.dopesheet import Dopesheet
+from mini.temporal.timeline import Timeline
 
 E = 4
 
 log = logging.getLogger(__name__)
 
 
-class ColorMLP(nn.Module):
-    def __init__(self):
+class ColorMLP(L.LightningModule):
+    def __init__(
+        self,
+        dopesheet: Dopesheet,
+        loss_criterion: LossCriterion,
+        regularizers: list[RegularizerConfig],
+    ):
         super().__init__()
+        # Store configuration
+        self.loss_criterion = loss_criterion
+        self.regularizers = regularizers
+        self.timeline = Timeline(dopesheet)
+
+        # Store latents for regularization via hooks
+        self.current_latents: Tensor | None = None
+
```

Comment:
I'm unfamiliar with Lightning. Is it usual to have these as part of the model? They seem to be more related to training than to the model itself.